rotkehlchen/exchanges/coinbase.py

```diff
@@ -702,11 +715,118 @@ def query_online_deposits_withdrawals(
             movement = self._deserialize_asset_movement(raw_movement)
             # limit coinbase deposit/withdrawals in the requested time range
             #  here since there is no argument in the API call
-            if movement and movement.timestamp >= start_ts and movement.timestamp <= end_ts:
+            if movement and start_ts <= movement.timestamp <= end_ts:
                 movements.append(movement)
 
         return movements
 
+    def _deserialize_ledger_action(self, raw_data: Dict[str, Any]) -> Optional[LedgerAction]:
+        """Processes a single transaction from coinbase and deserializes it
+
+        Can log error/warning and return None if something went wrong at deserialization
+        This method can raise:
+        - DeserializationError
+        - UnknownAsset
+        - UnsupportedAsset
+        """
+        try:
+            if raw_data['status'] != 'completed':
+                return None
+            payout_date = raw_data.get('payout_at', None)
+            if payout_date:
+                timestamp = deserialize_timestamp_from_date(payout_date, 'iso8601', 'coinbase')
+            else:
+                timestamp = deserialize_timestamp_from_date(
+                    raw_data['created_at'],
+                    'iso8601',
+                    'coinbase',
+                )
+            if 'type' in raw_data:
+                # The parent method filtered with 'from' attribute, so it is from another user.
+                # https://developers.coinbase.com/api/v2?python#transaction-resource
+                action_type = LedgerActionType.INCOME
+                if raw_data.get('type', '') not in ['send', 'inflation_reward']:
```

Comment:
use immutable type when possible

```suggestion
                if raw_data.get('type', '') not in ('send', 'inflation_reward'):
```